sign/rpmgensig.c

```diff
@@ -276,7 +282,26 @@ static int runGPG(sigTarget sigt, const char *sigfile)
     rpmPopMacro(NULL, "__plaintext_filename");
     rpmPopMacro(NULL, "__signature_filename");
 
+    /* The child GPG process may terminate without ever opening the pipe (such
+     * as when the key is expired), causing our parent process to get forever
+     * stuck on the Fopen() call below (see fifo(7) for details).  To fix that,
+     * we need to let the underlying open(2) unblock when SIGCHLD is received,
+     * by registering a no-op handler, so here we go: */
```

Comment:
On another purely stylistic note, this comment is on the verbose side. Some general guidelines on the topic:
- use comments sparingly, subtle and non-obvious things (this certainly classifies)
- try to stick to one-liners for in-line comments, unless it's something *really* special
- the preferred place for comments is at head of functions, and longer comments are okay there
- ...but story-telling belongs to commit messages (where OTOH this sort of background info is golden)

The kernel coding style differs in various aspects and some things just don't apply, but in general it's usually excellent advice and on comments it's totally spot-on in rpm too:
https://www.kernel.org/doc/html/v4.10/process/coding-style.html#commenting

Reply:
Wow, that's great feedback, thanks. I'll definitely check out the link.